lib/srv/reexec.go

```diff
@@ -662,6 +657,11 @@ func RunNetworking() (errw io.Writer, code int, err error) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	// Unlock OS thread (if locked by PAM) so that goroutines below inherit
+	// thread-state related to PAM, e.g. pam_namespace. See lib/pam/pam.go
+	// for details on why we lock the OS thread on init.
```

Comment:
`man pam_namespace` explicitly says "The module must not be called from multithreaded processes." so I'm not really sure what it means to "inherit thread state related to PAM".

Reply:
I'm not really sure how to understand it either, but sounds like this solution might have some hidden side effects since it's being explicitly warned against, thanks for catching that in the man page.

If we handle networking requests sequentially, the main downside is that dialing a TCP address with significant connection latency would prevent new requests, while tcp/unix listeners and unix dials should be quite fast. Should we just wrap the TCP dial handling in a goroutine alone? The only other solution I can think of would be to separate the subprocesses back out, or having the networking process create additional subprocesses when needed.

Reply:
What does openssh do? That's the usual guideline when dealing with the low level system integration 🤣

Reply:
I tried looking through the OpenSSH source code but I don't have enough C comprehension to understand how it managed to get around this multithreading issue. My current understanding is that the daemon creates a PAM auth thread for each user session, and any exec/forwarding requests within that session are ran with the PAM auth context. It's unclear to me how exactly that auth context is applied to calls to bind. Could the pam_namespace C library be wrapping it somehow? If you have more C experience I'd appreciate your insight.

WDYT about having the request handling loop run in a goroutine? Then we can have a bind-handling loop in the main thread which handles bind requests from the request handling goroutine? (Assuming bind is the only thing necessary from within the pam auth context)

Any other suggestions would be appreciated 😃